File: Template Method/PHP/TwoFactorPhone.php
<?php

require_once "TwoFactorAuthentication.php";

class TwoFactorPhone extends TwoFactorAuthentication {
    public function getData(string $username) {
        echo "Getting phone data for {$username}\n";
    }

    public function sendNotification(string $username) {
        echo "Sending sms to {$username}\n";
    }
    
    public function match(string $username) {
        echo "Matching sms code from {$username}\n";
    }
}

File: Template Method/Python 3/TwoFactorPhone.py
from TwoFactorAuthentication import TwoFactorAuthentication

class TwoFactorPhone(TwoFactorAuthentication):
  def getData(self, username):
    print('Getting phone number data for {}'.format(username))

  def sendNotification(self, username):
    print('Sending SMS notification to {}'.format(username))

  def match(self, username):
    print('Matching {} with the SMS code'.format(username))
